algorithms/linfa-ftrl/benches/ftrl.rs

```diff
@@ -0,0 +1,55 @@
+use linfa::prelude::Predict;
+use linfa::traits::FitWith;
+use linfa::{Dataset, DatasetBase, ParamGuard};
+use linfa_ftrl::FTRL;
+use ndarray::{Array1, Array2};
+use ndarray_rand::{
+    rand::distributions::Uniform, rand::rngs::SmallRng, rand::SeedableRng, RandomExt,
+};
+
+fn iai_fit_and_predict_without_prior_model() {
+    let mut rng = SmallRng::seed_from_u64(42);
+    let params = FTRL::params();
+    for nfeatures in (10..100).step_by(10) {
+        for size in (1_000..10_000).step_by(2_000) {
+            let dataset = get_dataset(&mut rng, size, nfeatures);
+            let model = params.fit_with(None, iai::black_box(&dataset)).unwrap();
+            model.predict(iai::black_box(dataset));
+        }
+    }
+}
```

Comment:
I'm not too familiar with `iai`, but this benchmark only counts the `fit_with` and `predict` calls right? It shouldn't be including the `get_dataset` call in the measurement.

Reply:
I'm only discovering this crate, but from what I can see in their documentation, specifically `Con: Iai cannot exclude setup code from the measurements, where Criterion-rs can.`, it does include everything. On the side note, I can rewrite benchmark with Criterion.

Reply:
Better to rewrite in Criterion then. Iai seems to lack a lot of features and isn't being actively developed.